src/async_impl/response.rs

```diff
@@ -421,6 +421,14 @@ impl From<Response> for Body {
     }
 }
 
+/// A `Response` can be transformed into a classic http hyper body response.
+impl From<Response> for http::Response<hyper::Body> {
+    fn from(r: Response) -> http::Response<hyper::Body> {
+        let (parts, body) = r.res.into_parts();
+        http::Response::from_parts(parts, hyper::Body::wrap_stream(body))
```

Comment:
I suppose we could give the body without wrapping it prematurely, and then anyone needing to unify it could wrap it themselves. If they didn't need to, we could save them a boxed trait object.

Is there any reason we shouldn't do that?

Reply:
Sounds good I just don't know how as the body returned by `into_parts` has the private `Decoder` type so we need to wrap it some way right? If this where a method I suppose we could return `http::Response<impl Stream<..>>`. What do you think?

Reply:
In at least the blocking implementation, I'd like to use a similar conversion trait for a simple `Vec<u8>`. See how `ureq` provides a few of them:

https://github.com/algesten/ureq/blob/916ffbffbe098d28d9ac142054484fd71423d9a6/src/http_interop.rs#L55-L129 

Seems there's also #1258, #1483, #1612 to track and implement (something) like this. The latter being a PR implementing the same trait in `src/async_impl/response.rs` but for anything with a `From<Decoder>`. @seanmonstar is one of these PRs "superseding" the other? Can I help gain some traction on any of these?